scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java

```diff
@@ -458,10 +455,39 @@ private void createFolderHierarchySequentially(
 
         if (isDebugEnabled) {
             long timeToCreateHierarchySequentially = System.currentTimeMillis() - startTime;
-            logger.debug("Creating hierarchy sequentially required " + timeToCreateHierarchySequentially + " ms");
+            logger.debug(
+                    "Executing TaskProActiveDataspaces#createFolderHierarchySequentially has taken " + timeToCreateHierarchySequentially + " ms");
         }
     }
 
+    private void createFolderHierarchy(boolean isDebugEnabled, DataSpacesFileObject fileObject,
+            DataSpacesFileObject target) throws FileSystemException {
+
+        FileType fileObjectType = fileObject.getType();
+
+        if (FileType.FOLDER.equals(fileObjectType)) {
+            if (isDebugEnabled) {
+                logger.debug("Creating folder " + target.getRealURI());
+            }
+
+            target.createFolder();
+        } else if (FileType.FILE.equals(fileObjectType)) {
+            DataSpacesFileObject parent = target.getParent();
+
+            if (isDebugEnabled) {
+                logger.debug("Creating folder " + parent.getRealURI());
+            }
+
+            parent.createFolder();
+        }
+    }
+
+    private boolean isCreateFolderHierarchySequentiallyEnabled() {
+        String property = System.getProperty(PA_NODE_DATASPACE_CREATE_FOLDER_HIERARCHY_SEQUENTIALLY);
+
+        return property == null || "true".equalsIgnoreCase(property);
+    }
+
```

Comment:
By default the property is not defined. In such a case I want the create folder hierarchy sequentially feature to be enabled. The property is there to have a mean to disable this last  feature.